ext/standard/array.c

```diff
@@ -4189,19 +4167,19 @@ PHP_FUNCTION(array_column)
 		ZEND_HASH_FOREACH_VAL(input, data) {
 			ZVAL_DEREF(data);
 
-			if (!column) {
+			if (column_is_null) {
 				Z_TRY_ADDREF_P(data);
 				colval = data;
-			} else if ((colval = array_column_fetch_prop(data, column, &rv)) == NULL) {
+			} else if ((colval = array_column_fetch_prop(data, column_str, column_long, &rv)) == NULL) {
 				continue;
 			}
 
 			/* Failure will leave keyval alone which will land us on the final else block below
 			 * which is to append the value as next_index
 			 */
-			if (index) {
+			if (!index_is_null) { // This is useless
```

Comment:
I'm pretty confused at to what this if block is attempting as even before the condition was always true.

Reply:
Yeah, this code is dead, I went ahead and dropped it in master.